src/test/throttler/stack.spec.ts

```diff
@@ -0,0 +1,47 @@
+import * as chai from "chai";
+import Stack from "../../throttler/stack";
+
+const { expect } = chai;
+
+describe("Stack", () => {
+  it("should have default name of stack", () => {
+    const stack = new Stack({});
+    expect(stack.name).to.equal("stack");
+  });
+
+  it("should be first-in-last-out", () => {
+    const stack = new Stack({});
+    expect(stack.hasWaitingTask()).to.equal(false);
+
+    stack.total++;
```

Comment:
Again, use the class like you _actually_ would to test it. Don't just set internal state.